mediator/esmFldsExchange_coastal_mod.F90

```diff
@@ -140,11 +148,25 @@ subroutine esmFldsExchange_coastal_advt(gcomp, phase, rc)
     ! to med: masks from components
     !----------------------------------------------------------
     call addfld_from(compocn, 'So_omask')
+    call addfld_from(compice, 'Si_imask')
 
     !----------------------------------------------------------
     ! to med: frac from components
     !----------------------------------------------------------
     call addfld_to(compatm, 'So_ofrac')
+          
+    if (coastal_attr%ice_present .and. coastal_attr%atm_present) then
+         call addfld_from(compice , 'Si_ifrac')
+         call addfld_to(compatm   , 'Si_ifrac')
+
+    end if
+
+    if (coastal_attr%ice_present .and. coastal_attr%ocn_present) then
+    !      call addfld_from(compocn, 'So_omask')
```

Comment:
@SmithJos13 Please remove the section related to ocean mask if it is not required.